### PR TITLE
Add current user to exception

### PR DESCRIPTION
**Pull request by jforgue**

### CODE REVIEW

**Comment by jforgue** [comment]
According to exception_notification's read me this should will show up in the email as a new section titled "Data", and should look something like this.
`------------------------------- Data:

* data: {:current_user=>
  #<User:0x007ff03c0e5860
   id: 3,
   email: "jane.doe@example.com", # etc...`

**Comment by jforgue** [comment]
I'm not sure how to test an exception, but I did check that `request.env["exception_notifier.exception_data"]` was assigned in the view.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -4,11 +4,17 @@ class ApplicationController < ActionController::Base
...
+  before_action :prepare_exception_notifier
 
   helper_method :format_datetime
   
   private
 
+  def prepare_exception_notifier
+    request.env["exception_notifier.exception_data"] = {
+        :current_user => current_user
+    }
```
Comment (dfaulken) [inline app/controllers/application_controller.rb]: Two concerns here:

1. Does this hash exist before? Do we want to merge rather than overwrite?
2. Hash rockets in flight

Comment (jforgue) [reply app/controllers/application_controller.rb]: If I don't initialize the hash then it returns nil in the view. So I don't think it exists before I assign it.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -4,11 +4,17 @@ class ApplicationController < ActionController::Base
...
 
   helper_method :format_datetime
   
   private
 
+  def prepare_exception_notifier
+    request.env['exception_notifier.exception_data'] = {
+      current_user: current_user
+    }
+  end
```
Comment (mboneil10) [inline app/controllers/application_controller.rb]: We typically at a newline at the end of methods.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -4,11 +4,17 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :authenticate_user!
   before_action :set_paper_trail_whodunnit
+  before_action :prepare_exception_notifier
```
Comment (mboneil10) [inline app/controllers/application_controller.rb]: Should this be a `before_action` or `before_filter`? I'm looking at the documentation and you've used `before_action`. Why this change?

Comment (jforgue) [reply app/controllers/application_controller.rb]: `before_filter` is no longer supported in rails.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (jforgue) [reply app/controllers/application_controller.rb]: OK

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (dfaulken) [reply app/controllers/application_controller.rb]: That may not necessarily be the case in production, however. Still best to merge rather than overwrite, I think.

Comment (jforgue) [reply app/controllers/application_controller.rb]: I mean the key `exception_notifier.exception_data` doesn't exist as far as I know, so I get a `NoMethodError (undefined method `merge' for nil:NilClass)`.

Comment (dfaulken) [reply app/controllers/application_controller.rb]: We can write code that handles it existing, and not overwriting if it does, and if it doesn't, initializing it with our value.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -7,13 +7,18 @@ class ApplicationController < ActionController::Base
...
   helper_method :format_datetime
-  
+
   private
 
   def prepare_exception_notifier
-    request.env['exception_notifier.exception_data'] = {
-      current_user: current_user
-    }
+    if request.env['exception_notifier.exception_data'].nil?
```
Comment (dfaulken) [inline app/controllers/application_controller.rb]: I would recommend something like

```ruby
hash ||= {}
hash.merge(stuff)
```
